autogen/agentchat/groupchat.py

```diff
@@ -52,11 +52,14 @@ def reset(self):
         """Reset the group chat."""
         self.messages.clear()
 
-    def append(self, message: Dict):
+    def append(self, message: Dict, speaker: Agent):
         """Append a message to the group chat.
         We cast the content to str here so that it can be managed by text-based
         model.
         """
+        # set the name to speaker's name if the role is not function or tool
+        if message["role"] != "function":
```

Comment:
Should it be `if message["role"] != "function" and message["role"] != "tool":` ? 

While both "function" and "tool" roles exist in different API version this check should be applied everywhere, not just here.

Reply:
It's OK when the role is tool. I modified the comment.

Reply:
What I mean is that in July when functioning calling was intoduced, the result of function call had to be added to caht messages using role "function". The current version of API suggests that "tool" role should be used instead. I think it is a bigger issue as everywhere where there can be a chat message with the result from function/tool, both roles must be accounted for

Reply:
That's true. #974 started adding support for tool. There are missing places to change. This PR addresses one of them.

Reply:
And I think that `and message["role"] != "tool":` should be added here to fix forward :)

Reply:
I did add a test for this case and found that we didn't need to add this. When the role is tool, the name field is not reserved, unlike the case of role=function